### ISSUE TITLE
stress: failed test in cockroach/sql/distsql/distsql.test: TestServer

### ISSUE THREAD

**Issue by mberhault** [cockroachdb/cockroach#7127]
Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/2d2fe00054e06e1081b24680463e4a7f951b1714

Stress build found a failed test:

```
=== RUN   TestServer
I160609 06:01:13.530513 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
W160609 06:01:13.534450 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node
I160609 06:01:13.555240 server/node.go:401  store store=0:0 ([]=) not bootstrapped
W160609 06:01:13.557556 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node
I160609 06:01:13.560386 storage/replica_command.go:1530  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 407070h1m18.060078972s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-06-09 06:01:13.560287487 +0000 UTC]
I160609 06:01:13.560945 server/node.go:327  **** cluster {aecae1dd-101c-4cc7-bab2-8b7233a6eb78} has been created
I160609 06:01:13.560975 server/node.go:328  **** add additional nodes by specifying --join=127.0.0.1:43261
W160609 06:01:13.560993 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node
I160609 06:01:13.561708 server/node.go:414  initialized store store=1:1 ([]=): {Capacity:10365558784 Available:8198770688 RangeCount:0}
I160609 06:01:13.561737 server/node.go:302  node ID 1 initialized
I160609 06:01:13.562531 storage/stores.go:287  read 0 node addresses from persistent storage
I160609 06:01:13.562594 server/node.go:535  connecting to gossip network to verify cluster ID...
I160609 06:01:13.562643 storage/split_queue.go:100  splitting range=1 [/Min-/Max) at keys [/Table/11/0 /Table/12/0 /Table/13/0 /Table/14/0]
I160609 06:01:13.562798 server/node.go:556  node connected via gossip and verified as part of cluster {"aecae1dd-101c-4cc7-bab2-8b7233a6eb78"}
I160609 06:01:13.562831 server/node.go:355  [node=1] Started node with [[]=] engine(s) and attributes []
I160609 06:01:13.562853 server/server.go:417  starting https server at 127.0.0.1:35171
I160609 06:01:13.562863 server/server.go:418  starting grpc/postgres server at 127.0.0.1:43261
I160609 06:01:13.564086 storage/replica_command.go:2077  initiating a split of range=1 [/Min-/Max) at key /Table/11
I160609 06:01:13.566186 server/updates.go:156  No previous updates check time.
I160609 06:01:13.570019 storage/replica_command.go:1530  range 2: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 407070h1m18.16126335s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-06-09 06:01:13.569978539 +0000 UTC]
I160609 06:01:13.570138 storage/replica_command.go:2077  initiating a split of range=2 [/Table/11-/Max) at key /Table/12
I160609 06:01:13.591453 storage/replica_command.go:1530  range 3: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 407070h1m18.16126335s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-06-09 06:01:13.591352995 +0000 UTC]
I160609 06:01:13.591577 storage/replica_command.go:2077  initiating a split of range=3 [/Table/12-/Max) at key /Table/13
I160609 06:01:13.606596 storage/replica_command.go:1530  range 4: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 407070h1m18.16126335s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-06-09 06:01:13.606556661 +0000 UTC]
I160609 06:01:13.606718 storage/replica_command.go:2077  initiating a split of range=4 [/Table/13-/Max) at key /Table/14
I160609 06:01:13.655844 storage/replica_command.go:1530  range 5: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 407070h1m18.205516334s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-06-09 06:01:13.655791416 +0000 UTC]
I160609 06:01:13.655960 storage/split_queue.go:100  splitting range=5 [/Table/14-/Max) at keys [/Table/50/0]
I160609 06:01:13.656232 storage/replica_command.go:2077  initiating a split of range=5 [/Table/14-/Max) at key /Table/50
I160609 06:01:13.665169 stopper.go:352  draining; tasks left:
3      kv/txn_coord_sender.go:870
1      storage/queue.go:383
1      storage/intent_resolver.go:341
1      server/node.go:803
1      gossip/infostore.go:286
I160609 06:01:13.665599 stopper.go:352  draining; tasks left:
3      kv/txn_coord_sender.go:870
1      storage/queue.go:383
1      storage/intent_resolver.go:341
1      server/node.go:803
I160609 06:01:13.665678 stopper.go:352  draining; tasks left:
2      kv/txn_coord_sender.go:870
1      storage/queue.go:383
1      storage/intent_resolver.go:341
1      server/node.go:803
W160609 06:01:13.666520 storage/intent_resolver.go:318  [node=1,store=1,range=1] failed to resolve intents: node unavailable; try another peer
I160609 06:01:13.666555 stopper.go:352  draining; tasks left:
2      kv/txn_coord_sender.go:870
1      storage/queue.go:383
1      server/node.go:803
I160609 06:01:13.666598 stopper.go:352  draining; tasks left:
1      storage/queue.go:383
1      server/node.go:803
1      kv/txn_coord_sender.go:870
E160609 06:01:13.666767 client/txn.go:370  failure aborting transaction: node unavailable; try another peer; abort caused by: server/node.go:804: node 1 stopped
I160609 06:01:13.666863 stopper.go:352  draining; tasks left:
1      storage/queue.go:383
1      server/node.go:803
E160609 06:01:13.666979 storage/queue.go:446  [split] range=5 [/Table/14-/Max): error: storage/split_queue.go:103: unable to split range=5 [/Table/14-/Max) at key "/Table/50/0": storage/replica_command.go:2130: split at key /Table/50 failed: server/node.go:804: node 1 stopped
I160609 06:01:14.668592 /go/src/google.golang.org/grpc/clientconn.go:499  grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:43261: getsockopt: connection refused"; Reconnecting to "127.0.0.1:43261"
I160609 06:01:16.542607 /go/src/google.golang.org/grpc/clientconn.go:499  grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:43261: getsockopt: connection refused"; Reconnecting to "127.0.0.1:43261"
I160609 06:01:16.746595 http2_client.go:919  transport: http2Client.notifyError got notified that the client transport was broken EOF.
I160609 06:01:17.746864 /go/src/google.golang.org/grpc/clientconn.go:499  grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:43261: getsockopt: connection refused"; Reconnecting to "127.0.0.1:43261"
--- FAIL: TestServer (5.16s)
	leaktest.go:86: Leaked goroutine: goroutine 274 [select]:
		google.golang.org/grpc.(*Conn).resetTransport(0xc820b76c30, 0x0, 0x0, 0x0)
			/go/src/google.golang.org/grpc/clientconn.go:494 +0xa45
		google.golang.org/grpc.(*Conn).reconnect(0xc820b76c30, 0xc820b1c780)
			/go/src/google.golang.org/grpc/clientconn.go:532 +0xc4
		google.golang.org/grpc.(*Conn).transportMonitor(0xc820b76c30)
			/go/src/google.golang.org/grpc/clientconn.go:557 +0x16b
		google.golang.org/grpc.NewConn.func1(0xc820b76c30)
			/go/src/google.golang.org/grpc/clientconn.go:355 +0x1b5
		created by google.golang.org/grpc.NewConn
			/go/src/google.golang.org/grpc/clientconn.go:356 +0x4e3
```

Run Details:

```
74 runs so far, 0 failures, over 5s
157 runs so far, 0 failures, over 10s
241 runs so far, 0 failures, over 15s
321 runs so far, 0 failures, over 20s
405 runs so far, 0 failures, over 25s
485 runs so far, 0 failures, over 30s
566 runs so far, 0 failures, over 35s
649 runs so far, 0 failures, over 40s
729 runs so far, 0 failures, over 45s
809 runs so far, 0 failures, over 50s
890 runs so far, 0 failures, over 55s
973 runs so far, 0 failures, over 1m0s
1058 runs so far, 0 failures, over 1m5s
1141 runs so far, 0 failures, over 1m10s
1225 runs so far, 0 failures, over 1m15s
1307 runs so far, 0 failures, over 1m20s
1390 runs so far, 0 failures, over 1m25s
1475 runs so far, 0 failures, over 1m30s
1560 runs so far, 0 failures, over 1m35s
1644 runs so far, 0 failures, over 1m40s
1726 runs so far, 0 failures, over 1m45s
1808 runs so far, 0 failures, over 1m50s
1891 runs so far, 0 failures, over 1m55s
1974 runs so far, 0 failures, over 2m0s
2057 runs so far, 0 failures, over 2m5s
2144 runs so far, 0 failures, over 2m10s
2225 runs so far, 0 failures, over 2m15s
2305 runs so far, 0 failures, over 2m20s
2392 runs so far, 0 failures, over 2m25s
2470 runs so far, 0 failures, over 2m30s
2554 runs so far, 0 failures, over 2m35s
2633 runs so far, 0 failures, over 2m40s
2717 runs so far, 0 failures, over 2m45s
2799 runs so far, 0 failures, over 2m50s
2880 runs so far, 0 failures, over 2m55s
2960 runs so far, 0 failures, over 3m0s
3044 runs so far, 0 failures, over 3m5s
3125 runs so far, 0 failures, over 3m10s
3211 runs so far, 0 failures, over 3m15s
3294 runs so far, 0 failures, over 3m20s
3376 runs so far, 0 failures, over 3m25s
3459 runs so far, 0 failures, over 3m30s
3544 runs so far, 0 failures, over 3m35s
3626 runs so far, 0 failures, over 3m40s
3708 runs so far, 0 failures, over 3m45s
3792 runs so far, 0 failures, over 3m50s
3873 runs so far, 0 failures, over 3m55s
3957 runs so far, 0 failures, over 4m0s
4040 runs so far, 0 failures, over 4m5s
4123 runs so far, 0 failures, over 4m10s
4207 runs so far, 0 failures, over 4m15s
4287 runs so far, 0 failures, over 4m20s
4367 runs so far, 0 failures, over 4m25s
4449 runs so far, 0 failures, over 4m30s
4529 runs so far, 0 failures, over 4m35s
4613 runs so far, 0 failures, over 4m40s
4695 runs so far, 0 failures, over 4m45s
4780 runs so far, 0 failures, over 4m50s
4861 runs so far, 0 failures, over 4m55s
4944 runs so far, 0 failures, over 5m0s
5025 runs so far, 0 failures, over 5m5s
5109 runs so far, 0 failures, over 5m10s
5193 runs so far, 0 failures, over 5m15s
5277 runs so far, 0 failures, over 5m20s
5359 runs so far, 0 failures, over 5m25s
5447 runs so far, 0 failures, over 5m30s
5531 runs so far, 0 failures, over 5m35s
5614 runs so far, 0 failures, over 5m40s
5695 runs so far, 0 failures, over 5m45s
5782 runs so far, 0 failures, over 5m50s
5868 runs so far, 0 failures, over 5m55s
5949 runs so far, 0 failures, over 6m0s
5998 runs completed, 1 failures, over 6m3s
FAIL
```
Please assign, take a look and update the issue accordingly.

**Comment by RaduBerinde** [comment]
Might be the same with #7117

**Comment by tamird** [comment]
Fixed in #7241.
